internal/nlwrap/nlwrap_linux.go

```diff
@@ -0,0 +1,172 @@
+// Package nlwrap wraps vishvandanda/netlink functions that may return EINTR.
+//
+// A Handle instantiated using [NewHandle] or [NewHandleAt] can be used in place
+// of a netlink.Handle, it's a wrapper that replaces methods that need to be
+// wrapped. Functions that use the package handle need to be called as "nlwrap.X"
+// instead of "netlink.X".
+//
+// The wrapped functions currently return EINTR when NLM_F_DUMP_INTR flagged
+// in a netlink response, meaning something changed during the dump so results
+// may be incomplete or inconsistent.
+//
+// At present, the possibly incomplete/inconsistent results are not returned
+// by netlink functions along with the EINTR. So, it's not possible to do
+// anything but retry. After maxAttempts the EINTR will be returned to the
+// caller.
+package nlwrap
+
+import (
+	"context"
+	"errors"
+
+	"github.com/containerd/log"
+	"github.com/vishvananda/netlink"
+	"github.com/vishvananda/netns"
+	"golang.org/x/sys/unix"
+)
+
+// Arbitrary limit on max attempts at netlink calls if they are repeatedly interrupted.
+const maxAttempts = 5
+
+type Handle struct {
+	*netlink.Handle
+}
+
+func NewHandle(nlFamilies ...int) (Handle, error) {
+	nlh, err := netlink.NewHandle(nlFamilies...)
+	if err != nil {
+		return Handle{}, err
```

Comment:
Curious; should we return a pointer here (and `nil` on error), similar to how the thing we wrap returns a pointer? https://github.com/moby/moby/blob/b6398f1d4ca0554cd852d55e8f8ff505348e42dc/vendor/github.com/vishvananda/netlink/handle_linux.go#L50-L52

Reply:
That's what I did initially, but @corhere pointed out ...

> My only complaint is the extra pointer indirection. Since `nlwrap.Handle` is effectively a reference type, a `*nlwrap.Handle` is laid out in memory identically to a `**netlink.Handle`. Passing around `nlwrap.Handle` values by-value would be more efficient
> It's a reference type in the same vein as slices -- a value which carries a reference

Reply:
Ah, right, hm, yeah, I was trying to see if we could make it a full drop-in replacement